resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventOutputImpl.java

```diff
@@ -107,7 +107,19 @@ protected void close(boolean flushBeforeClose)
             ResteasyAsynchronousResponse asyncResponse = asyncContext.getAsyncResponse();
             if (asyncResponse != null)
             {
-               asyncResponse.complete();
+               try {
+                  asyncResponse.complete();
+               } catch(RuntimeException x) {
+                  Throwable cause = x;
+                  while(cause.getCause() != null)
```

Comment:
cause.getCause() != null && cause.getCause() != cause

Reply:
That really happens?